### PR TITLE
Adds Rubocop 🚔

### PR DESCRIPTION
**Pull request by ndizazzo**
... and opens a virtually un-reviewable PR!

Explicitly not decomposing enables/disable rules so it's _just_ the minimum required rubocop changes.  Those can follow for better visibility during review.

### CODE REVIEW

File: xcknife.gemspec [hunk 1]
```diff
@@ -11,19 +13,19 @@ Gem::Specification.new do |s|
...
-  s.files         = `git ls-files`.split($INPUT_RECORD_SEPARATOR).reject { |f| f =~ /^spec/} + ["TestDumper/TestDumper.dylib"]
+  s.files         = `git ls-files`.split($INPUT_RECORD_SEPARATOR).reject { |f| f =~ /^spec/ } + ['TestDumper/TestDumper.dylib']
   s.bindir        = 'bin'
-  s.executables      = ['xcknife', 'xcknife-min', 'xcknife-test-dumper']
+  s.executables = %w[xcknife xcknife-min xcknife-test-dumper]
   s.require_paths = ['lib']
   s.test_files    = s.files.grep(%r{^(test|spec|features)/})
 
-  s.required_ruby_version = '>= 2.0.0'
+  s.required_ruby_version = '>= 2.6.0'
```
Comment (ndizazzo) [inline xcknife.gemspec:28]: Rubocop doesn't support less than 2.4, and complains that this mismatches otherwise.

File: bin/xcknife-min [hunk 2]
```diff
@@ -1,31 +1,28 @@
-#!/usr/bin/env ruby
-# encoding: utf-8
+# frozen_string_literal: true
 
+#!/usr/bin/env ruby
```
Comment (segiddins) [inline bin/xcknife-min:3]: this needs to be line 1

File: bin/xcknife-test-dumper [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
 #!/usr/bin/env ruby
```
Comment (segiddins) [inline bin/xcknife-test-dumper:3]: needs to stay line 1

File: lib/xcknife/xctool_cmd_helper.rb [hunk 4]
```diff
@@ -25,12 +29,18 @@ def xctool_only_arguments_for_a_partition_set(partition_set)
...
 
     # only-testing is available since Xcode 8
+    # rubocop:disable Metrics/CyclomaticComplexity
     def xcodebuild_only_arguments(single_partition, meta_partition = nil)
       only_targets = if meta_partition
-        single_partition.keys.to_set & meta_partition.flat_map(&:keys).group_by(&:to_s).select{|_,v| v.size == 1 }.map(&:first).to_set
-      else
-        Set.new
-      end
+                       single_partition.keys.to_set & meta_partition
```
Comment (segiddins) [inline lib/xcknife/xctool_cmd_helper.rb:35]: extract the second expression to a temp var so its more readable?

File: lib/xcknife/stream_parser.rb [hunk 5]
```diff
@@ -88,24 +92,27 @@ def compute_shards_for_partitions(test_time_for_partitions)
...
+        next unless analyzer.test_class?(target_name, result.className)
+
         times_for_target_class[target_name][result.className] += (result.totalDuration * 1000).ceil
       end
 
       extrapolate_times_for_current_events(analyzer, times_for_target_class) if current_events
       hash_partitions(times_for_target_class)
     end
 
+    # rubocop:disable Metrics/CyclomaticComplexity
```
Comment (justinseanmartin) [inline lib/xcknife/stream_parser.rb:111]: Alternative to this is configuring the threshold higher. Not sure how far off this method is compared to the configuration.

**Comment by justinseanmartin** [comment]
Approval is conditional on fixing the shebangs

File: lib/xcknife/stream_parser.rb (hunk 5 above)
Comment (ndizazzo) [reply lib/xcknife/stream_parser.rb:111]: In some cases it was really high (21/7), I wouldn't want to bump it in the rules as I think seeing these exceptions is more actionable. I think the right thing is to decompose the logic in follow-up PRs.

**Pull request merged**